WooCommerce/WooCommerceTests/ViewRelated/Orders/Review Order/ReviewOrderViewModelTests.swift

```diff
@@ -33,46 +34,97 @@ class ReviewOrderViewModelTests: XCTestCase {
 
         // Then
+        let aggregateItem = viewModel.aggregateOrderItems.first!
+        let productCellModel = viewModel.productDetailsCellViewModel(for: aggregateItem)
         XCTAssertEqual(productCellModel.hasAddOns, false)
     }
 
     func test_productDetailsCellViewModel_returns_correct_hasAddOns_if_view_model_receives_showAddOns_as_true_and_there_are_valid_addons() {
         // Given
         let addOnName = "Test"
         let itemAttribute = OrderItemAttribute.fake().copy(name: addOnName)
-        let item = OrderItem.fake().copy(productID: productID, attributes: [itemAttribute])
+        let item = OrderItem.fake().copy(productID: productID, quantity: 1, attributes: [itemAttribute])
         let order = Order.fake().copy(siteID: siteID, status: .processing, items: [item])
         let addOn = ProductAddOn.fake().copy(name: addOnName)
         let product = Product().copy(productID: productID, addOns: [addOn])
 
         // When
         let viewModel = ReviewOrderViewModel(order: order, products: [product], showAddOns: true, storageManager: storageManager)
-        let productCellModel = viewModel.productDetailsCellViewModel(for: item)
 
         // Then
+        let aggregateItem = viewModel.aggregateOrderItems.first!
+        let productCellModel = viewModel.productDetailsCellViewModel(for: aggregateItem)
         XCTAssertEqual(productCellModel.hasAddOns, true)
     }
+
+    func test_productSection_contains_only_non_refunded_items() {
+        // Given
+        let productID2: Int64 = 335
+        let itemID1: Int64 = 134
+        let itemID2: Int64 = 432
+
+        let product1 = Product().copy(productID: productID)
+        let product2 = Product().copy(productID: productID2)
+
+        let item1 = OrderItem.fake().copy(itemID: itemID1, productID: product1.productID, quantity: 1)
+        let item2 = OrderItem.fake().copy(itemID: itemID2, productID: product2.productID, quantity: -1)
+        let order = Order.fake().copy(siteID: siteID, orderID: orderID, status: .processing, items: [item1, item2])
+
+        let itemRefund = OrderItemRefund.fake().copy(itemID: item2.itemID, productID: item2.productID)
+        let refund = Refund.fake().copy(orderID: orderID, siteID: siteID, items: [itemRefund])
+        insert(refund)
+
+        // When
+        let viewModel = ReviewOrderViewModel(order: order, products: [product1, product2], showAddOns: false, storageManager: storageManager)
+        viewModel.configureResultsControllers {}
+
+        // Then
+        let productSection = viewModel.sections.first(where: { $0.category == .products })
+        XCTAssertNotNil(productSection)
+        let productRow = productSection?.rows.first(where: {
+            if case .orderItem(let item) = $0,
+               item.productID == productID {
+                return true
+            }
+            return false
+        })
+        XCTAssertNotNil(productRow)
```

Comment:
Thank you for adding this test! 🎉 

I might be mistaken, but I think this only checks that the non-refunded item is in the Products section, right? It doesn't check to make sure the refunded item **isn't** there.

What do you think of replacing this check (or adding another check) like this:

```
        let refundedProductRow = productSection?.rows.first(where: {
            if case .orderItem(let item) = $0,
               item.productID == productID2 {
                return true
            }
            return false
        })
        XCTAssertNil(refundedProductRow)
```

Reply:
That makes better sense! Let me update that 😄